tests/test_oidc.py

```diff
@@ -0,0 +1,21 @@
+import pytest
+from controlpanel.oidc import StateMismatchHandler
+from unittest.mock import Mock
+
+
+@pytest.mark.parametrize(
+    "email, success_url",
+    [
+        ("", "/"),
+        ("example@justice.gov.uk", "/tools/"),
+    ],
+)
+def test_success_url(users, email, success_url):
+    request = Mock()
```

Comment:
For future reference, there is a fixture you can use to create a request object https://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory